src/main/java/macrobase/analysis/pipeline/stream/MBMultiInputStream.java

```diff
@@ -0,0 +1,41 @@
+package macrobase.analysis.pipeline.stream;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MBMultiInputStream<T> extends MBStream<T> {
+    private final List<MBStream<T>> inputs;
+    private int idx = 0;
+
+    public MBMultiInputStream(List<MBStream<T>> inputs) {
+        this.inputs = inputs;
+    }
+
+    public MBMultiInputStream() {
+        this(new ArrayList<>());
+    }
+
+    public void addStream(MBStream<T> input) {
+        inputs.add(input);
+    }
+
+    @Override
+    public List<T> drain(int maxElements) {
```

Comment:
Can we add a test for streaming usage of this class? Getting streaming mini-batch draining to work seems tricky.